core/codegen/src/attribute/route/parse.rs

```diff
@@ -209,9 +212,23 @@ impl Route {
             })
             .collect();
 
+        let doc_comment = handler
+            .attrs
+            .iter()
+            .filter(|a| a.path.is_ident("doc"))
+            .filter_map(|attr| attr.parse_meta().ok())
+            .filter_map(|meta| if let NameValue(MetaNameValue { lit: Str(string), .. }) = meta {
+                Some(string.value())
+            } else {
+                None
+            })
+            .flat_map(|string| string.split('\n').map(|s| s.trim().to_owned()).collect::<Vec<_>>())
+            .fold(String::new(), |acc, elem| acc + &elem + "\n");
+        let doc_comment = (!doc_comment.is_empty()).then(|| doc_comment);
```

Comment:
Devise has utilities to parse attributes. Something like this should work:
```suggestion
        let docstring = String::from_attrs("doc", &handler.attrs)?.join("\n");
```

Do we actually need the new line, though? Or does it automatically get inserted when the docstring is converted into an attribute?

Reply:
I want to later detect whether the first n lines start with a pound sign, and then use those as a title attribute, so the newline may not (yet) be lost at this stage.

Reply:
No, it simply gives you the raw string literal value.

I think we should preserve the docstring as-is, for now. Any muxing can be done at a later stage or when we start making semantic sense of the string. I believe the string literals will already have newlines; I'm not sure there can be more than one. I would play with this. Try routing out some single line, multi line docstrings using `///` and using a raw attribute. mu